File: Java/1200. Minimum Absolute Difference.java
class Solution {
    public List<List<Integer>> minimumAbsDifference(int[] arr) {
        List<List<Integer>> ans = new ArrayList<>();
        int min = 10001, l = arr.length;

        Arrays.sort(arr);
        
        for(int i = 0; i < l - 1; i++) min = Math.min(min, arr[i + 1] - arr[i]);
        for(int i = 0; i < l - 1; i++){
            if(arr[i + 1] - arr[i] == min) ans.add(Arrays.asList(arr[i], arr[i + 1]));
        }

        return ans;
    }
}

File: Python/1200. Minimum Absolute Difference.py
class Solution:
    def minimumAbsDifference(self, arr: List[int]) -> List[List[int]]:
        l, m, res = len(arr), 10001, []
        
        arr.sort()

        for i in range(l - 1): m = min(m, arr[i + 1] - arr[i])
        for i in range(l - 1):
            if arr[i + 1] - arr[i] == m: res.append([arr[i], arr[i + 1]])
        
        return res
